src/main/java/org/spongepowered/api/world/extent/Extent.java

```diff
@@ -131,6 +132,28 @@ default Vector3i getHighestPositionAt(Vector3i position) {
     }
 
     /**
+     * Returns the y level that precipitation ends falling in the given column.
+     *
+     * <p>A value is still returned for columns in biomes which do not
+     * receive precipitation.</p>
+     *
+     * @return The y level that precipitation ends
+     */
+    int getPrecipitationLevel(int x, int z);
+
+    /**
+     * Returns the y level that precipitation ends falling in the given column.
+     *
+     * <p>A value is still returned for columns in biomes which do not
+     * receive precipitation.</p>
+     *
+     * @return The y level that precipitation ends
+     */
+    default int getPrecipitationLevel(Vector2i column) {
```

Comment:
Alright, I'll add a vec2 method to getHighestYAt and a vec3 method for getPrecipitationLevel.